### PR TITLE
Johnny/navigation header

### PR DESCRIPTION
**Pull request by johnny-t06**
## Description
- Set up the file structure for the pages
- Implement bottom nav bar and header with mui components
- Added Tailwind Support 
- 
## Issue
Solve #10 except Error/Message popup

## Test
Mobile View visit to the pages `private/course`, `private/board`, `private/queue`, and `private/profile`
Edit: Pages are now nested into `private/course/[courseId]`
## Notes
The header is kinda janky but I spent a lot of time trying to figure out how to make the component accept:
- centered and left - aligned texts
- left/right icons and buttons

Getting the combinations of these were kinda tricky but maybe there's a better way? 

Also I didn't implement the error popup because it'll be a little easier to test as we implement our client rendering components + its an edge case popup

### CODE REVIEW

File: src/app/private/board/page.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import Header from "@components/Header";
+import { Button, Typography } from "@mui/material";
+
+const Page = () => {
+  return (
+    <div>
+      <Header
```
Comment (wkim10) [inline src/app/private/board/page.tsx]: I think it might be better to put Header components into layout.tsx

File: src/app/private/board/page.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        title="Board"
+        rightIcon={
+          <Button style={{ textTransform: "none" }}>
```
Comment (wkim10) [inline src/app/private/board/page.tsx]: nit: is there a way to use tailwind for this? i tried adding a class and doesn't seem to work cause mui but maybe there's a workaround?

File: src/app/globals.css [hunk 3]
```diff
@@ -16,15 +16,15 @@
   }
 }
 
-body {
+/* body {
```
Comment (wkim10) [inline src/app/globals.css]: nit: remove comment if not needed

File: src/app/components/NavBar.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import BottomNavigation from "@mui/material/BottomNavigation";
+import { BottomNavigationAction } from "@mui/material";
+import { SvgIconComponent } from "@mui/icons-material";
+
+interface INavBarItem {
+  label: string;
+  icon: React.ReactElement<SvgIconComponent>;
+  href: string;
+  // active: boolean;
```
Comment (wkim10) [inline src/app/components/NavBar.tsx]: nit: do we need this + what is active for?

File: src/app/private/board/page.tsx (hunk 1 above)
Comment (wkim10) [reply src/app/private/board/page.tsx]: oh wait unless we don't want it to display for nested pages/layouts, i just re-checked the figma

File: src/app/private/layout.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import HomeOutlinedIcon from "@mui/icons-material/HomeOutlined";
+import StickyNote2OutlinedIcon from "@mui/icons-material/StickyNote2Outlined";
+import PeopleAltOutlinedIcon from "@mui/icons-material/PeopleAltOutlined";
+import PersonOutlineOutlinedIcon from "@mui/icons-material/PersonOutlineOutlined";
+import { NavBarContainer } from "@components/container";
+import HeaderContainer from "@components/container/HeaderContainer";
```
Comment (wkim10) [inline src/app/private/layout.tsx]: nit: remove import if we don't need

File: src/app/private/board/page.tsx (hunk 1 above)
Comment (johnny-t06) [reply src/app/private/board/page.tsx]: Yeah I couldn't think of an elegant solution for a Layout component because each Tab (profile, board, queue, profile) and their nested pages have different headers

File: src/app/components/NavBar.tsx (hunk 4 above)
Comment (johnny-t06) [reply src/app/components/NavBar.tsx]: I was thinking to display a different (active) icon when the user is on that page/tab, but we can handle that next time.

File: src/app/private/board/page.tsx (hunk 1 above)
Comment (wkim10) [reply src/app/private/board/page.tsx]: yeah makes sense i think it looks good for now 👍

**Review by wkim10 — APPROVED** [review]
good job as always mr pm

File: src/app/components/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        {leftIcon}
+        <Typography
+          variant="h6"
+          component="div"
+          className={`flex-grow ${alignCenter ? "text-center" : "text-start"}`}
+        >
+          {title}
+        </Typography>
+
+        {alignCenter && !rightIcon ? <Box className="w-[40px]" /> : rightIcon}
```
Comment (wkim10) [inline src/app/components/Header.tsx:25]: a little confused about this logic, if alignCenter=false and rightIcon isn't given will rightIcon be null/undefined? i tried it and seems to be working but a little confused

File: src/app/hooks/useAuth.tsx [hunk 7]
```diff
@@ -15,7 +15,7 @@ const useAuth = () => {
   const onSignIn = useCallback(() => {
     const unsub = onAuthStateChanged(auth, (user) => {
       if (user) {
-        router.push("/course");
+        router.push("/private/course/cs160");
```
Comment (wkim10) [inline src/app/hooks/useAuth.tsx:18]: this is just temporary right? like we would normally just push to /course?

File: src/app/components/Header.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import {
+  AppBar,
+  Box,
+  Button,
```
Comment (wkim10) [inline src/app/components/Header.tsx]: nit: remove unneeded imports

File: src/app/components/Header.tsx (hunk 6 above)
Comment (johnny-t06) [reply src/app/components/Header.tsx:25]: My thought process is this: 
- If alignCenter == true and there is no right icon, we need to return an empty div/box.
- if alignCenter == true and there is a right icon, we can render the right icon.
- if alignCenter == false and there is no icon, it will go to the else statement of the ternary and try to render `{rightIcon}` which would be nothing.

File: src/app/hooks/useAuth.tsx (hunk 7 above)
Comment (johnny-t06) [reply src/app/hooks/useAuth.tsx:18]: Yes, thanks for pointing that out! We will change this once the userContext #9 is merged in.
